metal-stack/metal-api
# WIP: Persist user email who locked or reserved a machine in DB #303

When locking or reserving a specific machine, the metal-api additionally tracks the email address of the state change issuer.
This allows other users to directly contact the issuer in case they need additional information.

This PR requires additional changes in other repositories:

- metal-stack/metal-api _(pr tbd)_: tracking the issuer and provide access via api
- metal-stack/metal-go _(pr tbd)_: reflects the added issuer field of the api
- metal-stack/metalctl _(pr after merge)_: bump metal-go dependency (requires metal-go release)

As a new response field has been added to the api, this probably requires a minor update.

**How to test?**

_To use the updated metalctl, bump the metal-go version in `go.mod`, e.g. with `replace github.com/metal-stack/metal-go => ../metal-go`_

- assumes you have a running metal stack (e.g. mini-lab) and the fitting environment (`eval $(make dev-env)` from mini-lab)
- with new metalctl against old metal-api: `go run . machine describe $id` should work except issuer
- deploy this metal-api (`make mini-lab-push`)
- use the metalctl with a locally bumped metal-go version
- with new or old metalctl: `go run . machine lock $id`
- with new metalctl to see issue: `go run . machine describe $id`
